assets/app/views/directives/osc-routing.html

```diff
@@ -1,83 +1,216 @@
+    <div class="form-group">
+      <label for="host">Hostname</label>
+      <!-- regex, maxlength from k8s.io/kubernetes/pkg/util/validation/validation.go -->
+      <input
+          id="host"
+          class="form-control form-control-md"
+          type="text"
+          name="host"
+          ng-model="route.host"
+          ng-pattern="/^[a-z0-9]([-a-z0-9]*[a-z0-9])?(\.[a-z0-9]([-a-z0-9]*[a-z0-9])?)*$/"
+          ng-maxlength="253"
+          placeholder="www.example.com"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          aria-describedby="route-host-help">
+      <div>
+        <span id="route-host-help" class="help-block">
+          Public hostname for the route. If not specified, a hostname is generated.
+        </span>
+      </div>
+      <div class="has-error" ng-show="routeForm.host.$error.pattern && routeForm.host.$touched && !routingDisabled">
+        <span class="help-block">
+          Hostname must consist of lower-case letters, numbers, periods, and
+          hyphens. It must start and end with a letter or number.
+        </span>
       </div>
     </div>
-  </div><!-- /row -->
-  <div ng-class="fade" ng-show="route.customCerts">
-    <div class="row">
-      <div class="col-sm-offset-1 col-md-offset-1 col-md-2 col-sm-2">
-          <label>Certificate</label>
+
+    <!-- Path -->
+    <div class="form-group">
+      <label for="path">Path</label>
+      <input
+          id="path"
+          class="form-control form-control-md"
+          type="text"
+          name="path"
+          ng-model="route.path"
+          ng-pattern="/^\/.*$/"
+          ng-disabled="route.tls.termination === 'passthrough'"
+          placeholder="/"
+          autocorrect="off"
+          autocapitalize="off"
+          spellcheck="false"
+          aria-describedby="route-path-help">
+      <div>
+        <span id="route-path-help" class="help-block">
+          Path that the router watches to route traffic to the service.
+        </span>
+      </div>
+      <div class="has-error" ng-show="routeForm.path.$error.pattern && routeForm.path.$touched && !routingDisabled">
+        <span class="help-block">
+          Path must start with <code>/</code>
+        </span>
       </div>
-      <div class="col-md-9 col-sm-9">
-        <osc-file-input
-            name="certificate"
-            model="route.certificate"
-            required="route.customCerts"
-            >
-        </osc-file-input>
+      <div class="has-warning" ng-show="route.path && route.tls.termination === 'passthrough'">
+        <span class="help-block">
+          Path-based routing is not available when using passthrough TLS.
+        </span>
       </div>
-    </div><!-- /row -->
-    <div class="row">
-      <div class="col-sm-offset-1 col-md-offset-1 col-md-2 col-sm-2">
-          <label>Private Key</label>
+    </div>
+
+    <!-- Target Port -->
+    <div ng-if="route.portOptions.length" class="form-group">
```

Comment:
Yeah, this is something we were already doing. See https://github.com/openshift/origin/issues/4603#issuecomment-147251334 and https://github.com/openshift/origin/pull/4956#issuecomment-147741253